src/backend/distributed/utils/shard_utils.c

```diff
@@ -11,21 +11,50 @@
 
 #include "postgres.h"
 #include "utils/lsyscache.h"
-
+#include "distributed/metadata_cache.h"
 #include "distributed/relay_utility.h"
 #include "distributed/shard_utils.h"
 
 /*
- * GetShardLocalTableOid returns the oid of the shard from the given distributed relation
- * with the shardid.
+ * GetShardLocalTableOid returns the oid of the shard from the given distributed
+ * relation with the shardId.
+ */
+Oid
+GetShardLocalTableOid(Oid citusTableOid, uint64 shardId)
+{
+	const char *citusTableName = get_rel_name(citusTableOid);
+
+	Assert(citusTableName != NULL);
+
+	/* construct shard relation name */
+	char *shardRelationName = pstrdup(citusTableName);
+	AppendShardIdToName(&shardRelationName, shardId);
+
+	Oid citusTableSchemaOid = get_rel_namespace(citusTableOid);
+
+	Oid shardRelationOid = get_relname_relid(shardRelationName, citusTableSchemaOid);
+
+	return shardRelationOid;
+}
+
+
+/*
+ * GetOnlyShardOidOfReferenceTable returns OID of the one and only placement
+ * of the given reference table. Caller of this function must ensure that
+ * referenceTableOid is owned by a reference table.
  */
 Oid
-GetShardLocalTableOid(Oid distRelId, uint64 shardId)
+GetOnlyShardOidOfReferenceTable(Oid referenceTableOid)
```

Comment:
So this method returns the Oid of the reference table shard right?

If so we may consider renaming it to `GetShardLocalReferenceTableOid`.

Reply:
> So this method returns the Oid of the reference table shard right?

yes

> If so we may consider renaming it to GetShardLocalReferenceTableOid.

But we only have one & only shard for reference tables, `Local` doesn't make much sense to me in reference table context

Reply:
There is one shard per node. So if you have 10 worker nodes, there will be 10 shards in total, assuming the coordinator doesnt have it.

Here we return the Oid of the local shard of reference table rather than a shard that exists on another node. That is the reason the other method is named `GetShardLocalTableOid`. Hence I believe this should be named `GetShardLocalReferenceTableOid`.

Reply:
now I got what you mean, renaming it

Reply:
what do you think @SaitTalhaNisanci  ?

`GetShardLocalTableOid` -> `GetOidOfLocalShard`
`GetOnlyShardOidOfReferenceTable` -> `GetLocalPlacementOidOfReferenceTable`

Reply:
I think they should be consistent, one has shard, one has placement. 

Maybe `GetTableLocalShardOid` and `GetReferenceTableLocalShardOid` ? what do you think?

Reply:
> I think they should be consistent, one has shard, one has placement.

this is because `GetOidOfLocalShard` is to be called for a specific shard, but the other one is for a local placement of a shard